js/main.js

```diff
@@ -7,146 +37,75 @@ function getRandomInteger(min, max) {
-  author: {
-    avatar: avatars.getRandomItem(),
-  },
+const getPrice = () => {
+  getRandomInteger(PRICE.min, PRICE.max);
+};
 
-  offer: {
-    title: 'Аренда жилья в Токио',
-    address: '{{location.x}}, {{location.y}}',
-    price: getRandomInteger(1000, 10000),
-    type: housingType.getRandomItem(),
-    rooms: getRandomInteger(1, 3),
-    guests: getRandomInteger(1, 5),
-    checkin: checkinArray.getRandomItem(),
-    checkout: checkoutArray.getRandomItem(),
-    features: newArrayFeatures.slice(0, getRandomInteger(1, allFeatures.length)),
-    description: 'Чистое, светлое помещение для комфортного проживания с отличной инфраструктурой.',
-    photos: newArrayPhotos.slice(0, getRandomInteger(1, allPhotos.length)),
-  },
+const getGuests = () => {
+  getRandomInteger(GUESTS.min, GUESTS.max);
+};
 
-  location: {
-    lat: getRandomFloatInteger(35.65000, 35.70000, 5),
-    lng: getRandomFloatInteger(139.70000, 139.80000, 5),
-  },
+const getRooms = () => {
+  getRandomInteger(ROOMS.min, ROOMS.max);
 };
-ANNOUNCEMENT;
 
-function getAnnouncementArray (amount) {
-  const announcementArray = [];
-  for (let ii = 1; ii <= amount; ii++) {
-    announcementArray.push({
+const getOfferArray = (amount) => {
+  const offerArray = [];
+  for (let i = 1; i <= amount; i++) {
+    const CURRENT_LOCATION = getLocation();
+    offerArray.push({
       author: {
-        avatar: avatars.getRandomItem(),
+        avatar: `img/avatars/user0${getRandomInteger(1, 8)}.png`,
       },
 
       offer: {
         title: 'Аренда жилья в Токио',
-        address: '{{location.x}}, {{location.y}}',
-        price: getRandomInteger(1000, 10000),
-        type: housingType.getRandomItem(),
-        rooms: getRandomInteger(1, 3),
-        guests: getRandomInteger(1, 5),
-        checkin: checkinArray.getRandomItem(),
-        checkout: checkoutArray.getRandomItem(),
-        features: newArrayFeatures.slice(0, getRandomInteger(1, allFeatures.length)),
+        address: CURRENT_LOCATION,
+        price: getPrice(),
+        type: getRandomArrayElement(getShuffledArray(HOUSING_TYPES)),
+        rooms: getRooms(),
+        guests: getGuests(),
+        checkin: getRandomArrayElement(CHECKIN_ARRAY),
+        checkout: getRandomArrayElement(CHECKOUT_ARRAY),
+        features: getShuffledArray(FEATURES).slice(0, getRandomInteger(1, FEATURES.length)),
         description: 'Чистое, светлое помещение для комфортного проживания с отличной инфраструктурой.',
-        photos: newArrayPhotos.slice(0, getRandomInteger(1, allPhotos.length)),
-      },
-
-      location: {
-        lat: getRandomFloatInteger(35.65000, 35.70000, 5),
-        lng: getRandomFloatInteger(139.70000, 139.80000, 5),
+        photos: getShuffledArray(PHOTOS).slice(0, getRandomInteger(1, PHOTOS.length)),
+        location: CURRENT_LOCATION,
       },
     });
   }
-  return announcementArray;
-}
-announcementArray;
-getAnnouncementArray(10);
+  return offerArray;
+};
+const adsArray = getOfferArray(10);
```

Comment:
Раз уж решила использовать **offer** в функции, то и переменная должна называться соответственно)
```suggestion
const offersArray = getOfferArray(10);
```